Phase5/Search/FullTextSearch.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+
+namespace Phase5
+{
+    public class FullTextSearch : IFullTextSearch
+    {
+        private readonly IInvertedIndex _invertedIndex;
+        private readonly IDocsFileReader _docsFileReader;
+        private readonly ITokenizer _tokenizer;
+        private readonly IFilterApplier _filterApplier;
+
+        public FullTextSearch(IInvertedIndex invertedIndex, IDocsFileReader docsFileReader,ITokenizer tokenizer, IFilterApplier filterApplier)
+        {
+            _invertedIndex = invertedIndex;
+            _docsFileReader = docsFileReader;
+            _tokenizer = tokenizer;
+            _filterApplier = filterApplier;
+        }
+
+        public HashSet<string> FindCommandResult(string command)
+         {
+            LoadDocs();
+            var commandWords = SplitCommand(command);
+            return _filterApplier.Filter(commandWords);
+        }
+
+        private void LoadDocs()
```

Comment:
As I understand this part of code is initializing invertedIndex, so it's better to move it in that class or create a new one just for filling inverteIndexMap.